pgmpy/inference/base.py

```diff
@@ -80,3 +80,40 @@ def __init__(self, model):
             for factor in model.get_factors():
                 for var in factor.variables:
                     self.factors[var].append(factor)
+
+
+class ApproxInference:
+    """
+    Base class for the approximate inference algorithm.
+    Parameters
+    ----------
+    model: pgmpy.models.MarkovModel
+    We need a pairwise MarkovModel for Approx. inference.
+    """
+    def __init__(self, model):
+
+        # List of the variables in the present pairwise Markov Model
+        self.variables = []
+
+        # List of the cardinalities of the above variables in that order
+        self.cardinality = []
+
+        # List of all the factors that are present in the current model
+        self.factors = []
+
+        # List of all the factors values that are present in the current model
+        self.factor_values = []
+
+        # In a pairwise markov model, the order of the single node factors will define the cardinality order in the
+        # list. ( Following the way in which the data is provided in a UAI file )
```

Comment:
Please don't code things according to any file format. The inference class shouldn't be written how the values are provided in UAI file format. And please don't put UAI file again in the comments. Its misleading and makes the reader thinks that this class is specific to that file format.